src/plugins/navigation_embeddable/public/components/editor/navigation_embeddable_panel_editor_link.tsx

```diff
@@ -39,24 +42,85 @@ export const NavigationEmbeddablePanelEditorLink = ({
   parentDashboard?: DashboardContainer;
   dragHandleProps?: DraggableProvidedDragHandleProps;
 }) => {
+  const [dashboardError, setDashboardError] = useState<Error | undefined>();
   const parentDashboardTitle = parentDashboard?.select((state) => state.explicitInput.title);
   const parentDashboardId = parentDashboard?.select((state) => state.componentState.lastSavedId);
 
   const { value: linkLabel, loading: linkLabelLoading } = useAsync(async () => {
-    let label = link.label;
-    if (link.type === DASHBOARD_LINK_TYPE && !label) {
+    if (link.type === DASHBOARD_LINK_TYPE) {
       if (parentDashboardId === link.destination) {
-        label = parentDashboardTitle;
+        return link.label || parentDashboardTitle;
       } else {
-        const dashboard = await fetchDashboard(link.destination);
-        label = dashboard.attributes.title;
+        const dashboard = await fetchDashboard(link.destination)
+          .then((result) => {
+            setDashboardError(undefined);
+            return result;
+          })
+          .catch((error) => setDashboardError(error));
+        return (
+          link.label ||
+          (dashboard ? dashboard.attributes.title : DashboardLinkStrings.getDashboardErrorLabel())
+        );
       }
+    } else {
+      return link.label || link.destination;
     }
-    return label || link.destination;
   }, [link]);
 
+  const LinkLabel = useMemo(() => {
+    const labelText = (
+      <EuiFlexGroup tabIndex={0} gutterSize="s" responsive={false} wrap={false} alignItems="center">
+        <EuiFlexItem grow={false}>
+          <EuiIcon
+            type={dashboardError ? 'warning' : NavigationLinkInfo[link.type].icon}
+            color={dashboardError ? 'warning' : 'text'}
+            aria-label={
+              dashboardError
+                ? NavEmbeddableStrings.editor.panelEditor.getBrokenDashboardLinkAriaLabel()
+                : NavigationLinkInfo[link.type].type
+            }
+          />
+        </EuiFlexItem>
+
+        <EuiFlexItem
+          className={classNames('navEmbeddableLinkText', {
+            'navEmbeddableLinkText--noLabel': !link.label,
+          })}
+        >
+          <EuiSkeletonTitle
+            size="xxxs"
+            isLoading={linkLabelLoading}
+            announceLoadedStatus={false}
+            announceLoadingStatus={false}
```

Comment:
According to the [EUI docs](https://eui.elastic.co/#/display/skeleton#text), the `contentAriaLabel` is only necessary if we are announcing the loading status via the screen reader, which I removed with 

```typescript
announceLoadedStatus={false}
announceLoadingStatus={false}
```

I'm not entirely sure that this is the right move but, because of the way we are loading links **as they come** and not in a large bundle, the screenreader went crazy with "Link loaded" messages when I tried it out 😅 

Perhaps there's a way to get an overall accessibility review before the release? I think that could be super valuable :eyes:

Reply:
> Perhaps there's a way to get an overall accessibility review before the release? I think that could be super valuable, because I'm sure there's stuff we've missed 👀

Good idea. I've added an issue. https://github.com/elastic/kibana/issues/163802